src/plugins/dashboard/public/services/dashboard_content_management_service/lib/get_dashboard_state.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { pick } from 'lodash';
+import moment, { Moment } from 'moment';
+import { isFilterPinned } from '@kbn/es-query';
+import type { SavedObjectReference } from '@kbn/core/server';
+import { RefreshInterval, extractSearchSourceReferences } from '@kbn/data-plugin/public';
+
+import {
+  convertPanelMapToPanelsArray,
+  extractReferences,
+  generateNewPanelIds,
+} from '../../../../common';
+import type { DashboardAttributes } from '../../../../server';
+
+import { convertDashboardVersionToNumber } from './dashboard_versioning';
+import { DashboardSearchSource, GetDashboardStateProps } from '../types';
+import { dataService, embeddableService, savedObjectsTaggingService } from '../../kibana_services';
+import { LATEST_DASHBOARD_CONTAINER_VERSION } from '../../../dashboard_container';
+
+export const convertTimeToUTCString = (time?: string | Moment): undefined | string => {
+  if (moment(time).isValid()) {
+    return moment(time).utc().format('YYYY-MM-DDTHH:mm:ss.SSS[Z]');
+  } else {
+    // If it's not a valid moment date, then it should be a string representing a relative time
+    // like 'now' or 'now-15m'.
+    return time as string;
+  }
+};
+
+export const getDashboardState = async ({
```

Comment:
Why is this part of content management service instead of just a function that can be imported where needed? Does it need to be part of content management service?

Reply:
It does not need to be on content management service. I moved the module into the `dashboard_api` directory in ed68550a2d3f51657fb1414a8501c98b4b312a18.